Add java truststore configmap

mathianasj opened the pull request:
closes #52

mathianasj commented:
I need to add some readme updates, I should have marked this a draft

raffaelespazzoli commented on pkg/controller/cainjection/secret_controller.go:
```diff
@@ -58,6 +58,12 @@ func addSecretReconciler(mgr manager.Manager, r reconcile.Reconciler) error {
 			if oldSecretDest != newSecretDest {
 				return true
 			}
+
```
why did you have to change this file? from what I understand you were just adding a new controller...

mathianasj replied:
It looks like stuff from when I thought I had to go that direction made it on the wrong branch let me clean it up

raffaelespazzoli commented:
if we do this the way the key in the configmap representing the cabundle is going to be very OCP specific. This is good as a default, but can be find a way to let the user specify the key? Maybe an additional annotation?

raffaelespazzoli commented:
@mathianasj did you see this comment:
```
if we do this the way the key in the configmap representing the cabundle is going to be very OCP specific. This is good as a default, but can be find a way to let the user specify the key? Maybe an additional annotation?
```
thoughts?

mathianasj commented:
I am working on that now

mathianasj commented:
> @mathianasj did you see this comment:
> 
> ```
> if we do this the way the key in the configmap representing the cabundle is going to be very OCP specific. This is good as a default, but can be find a way to let the user specify the key? Maybe an additional annotation?
> ```
> 
> thoughts?

I have implemented this, let me know if you think the annotation name is appropriate, also added a unit test to validate it

raffaelespazzoli commented on README.md:
```diff
@@ -71,6 +72,17 @@ A such annotated secret looks like the following:
 
 The default password for these keystores is `changeme`. The password can be changed by adding the following optional annotation: `cert-utils-operator.redhat-cop.io/java-keystore-password: <password>`. The alias of the certificate inside the keystore is `alias`.
 
+### ConfigMaps
```
can you update the docs to reflect the new annotation available?

mathianasj replied:
added docs for annotations and defaults
